master/server.go

```diff
@@ -76,6 +85,24 @@ func (s *Server) RegisterExecutor(ctx context.Context, req *pb.RegisterExecutorR
 	}, nil
 }
 
+// ScheduleTask implements grpc interface. It works as follows
+// - receives request from job master
+// - queries resource manager to allocate resource and maps tasks to executors
+// - returns scheduler response to job master
+func (s *Server) ScheduleTask(ctx context.Context, req *pb.TaskSchedulerRequest) (*pb.TaskSchedulerResponse, error) {
+	// TODO: support running resource manager independently, and get resource snapshot via rpc.
+	snapshot := s.jobManager.resourceMgr.GetResourceSnapshot()
```

Comment:
Well , it seems we can get snapshot from executorManager directly, which is the implemention of resourceMgr.